src/useObservable.ts

```diff
@@ -68,28 +68,26 @@ export function useObservable<ObservableType extends Observable<any>, InitialVal
     cache.set(observable, entry as CacheRecord<ObservedValueOf<ObservableType>>)
   }
 
-  const store = useMemo(() => {
-    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-    const instance = cache.get(observable)!
-    return {
-      subscribe: (onStoreChange: () => void) => {
-        const subscription = instance.observable.subscribe(onStoreChange)
-        return () => {
-          subscription.unsubscribe()
-        }
-      },
-      getSnapshot: () => {
-        if (instance.error) {
-          throw instance.error
-        }
-        return instance.snapshot
-      },
-    }
-  }, [observable])
+  const subscribe = useCallback(
+    (onStoreChange: () => void) => {
+      const instance = cache.get(observable)!
+      const subscription = instance.observable.subscribe(onStoreChange)
+      return () => {
+        subscription.unsubscribe()
+      }
+    },
+    [observable],
+  )
 
   return useSyncExternalStore<ObservedValueOf<ObservableType>>(
-    store.subscribe,
-    store.getSnapshot,
+    subscribe,
+    () => {
+      const instance = cache.get(observable)!
+      if (instance.error) {
+        throw instance.error
+      }
+      return instance.snapshot
+    },
```

Comment:
Only the `subscribe` argument in `useSyncExternalStore` needs to be memoized, as it'll run teardown + setup every time it changes. The `getSnapshot` and `getServerSnapshot` arguments don't have to be memoized and the React Compiler uses less memory if we only memoize `subscribe` 🙌